meta_builds.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+#
+# This builds different stuff depending on the compiler:
+# gcc - valgrind, coverage
+# clang - asan, ubsan, scan-build
+# both - the two testbuild's NOTEST and NOFILE
+
+set -e
+
+if [ "$#" = "5" -a "$(echo $3 | grep -v 'makefile[.]')" = "" ]; then
+   echo "only run $0 for the regular makefile, early exit success"
+   exit 0
+fi
+
+function run_gcc() {
+   bash check_source.sh "CHECK_SOURCES" "$2" "$3" "$4" "$5"
+
+   make clean &>/dev/null
+
+   bash coverage.sh "COVERAGE" "$2" "$3" "$4" "$5"
+
+   make clean &>/dev/null
+
+   make CFLAGS="$2 $CFLAGS $4" EXTRALIBS="$5" test LTC_DEBUG=1 1>gcc_1.txt 2>gcc_2.txt
+
+   valgrind --error-exitcode=666 --leak-check=full --show-leak-kinds=all --errors-for-leak-kinds=all ./test 1>test_std.txt 2> test_err.txt
+
+   make clean &>/dev/null
+
+   make CFLAGS="-fsanitize=address -fno-omit-frame-pointer -static-libasan $2 $CFLAGS $4" EXTRALIBS="-lasan $5" test LTC_DEBUG=1 1>gcc_1.txt 2>gcc_2.txt
+   ASAN_OPTIONS=verbosity=1 ./test t ltm 1>test_std.txt 2> test_err.txt
```

Comment:
shouldn't ASAN go into `run_clang` ?

Reply:
I thought so too ... then I wasn't able to get the ASAN build done with clang which can be installed in xenial ... but it worked with gcc, so I thought let's give it a try...